lib/ddtrace/monkey.rb

```diff
@@ -18,6 +19,7 @@ module Monkey
       http: true,
       redis: true,
       grape: true,
+      faraday: true,
```

Comment:
Case scenario:
* I'm using faraday with Net::http adapter
* I do `Monkey.patch_all`
* both Faraday and http are instrumented so I get double spans (and metas) for each call
* users may want it to monitor Faraday overhead (i.e. middlewares) but others may not
* I think we should document that and provide a snippet to disable the underlying instrumentation (i.e. `Monkey.patch_all(http: false)`

Reply:
So do you think we should support 3 different scenarios?

* Only `net/http` instrumentation;
* Only `faraday` instrumentation;
* Both;

Is that right? Right now the user can easily disable the `faraday` integration, since the `Monkey` merely register our `Middleware` in the `faraday` registry, so its the client responsibility to enable/disable it for each `faraday` connection object. The second case is the problem, because we rely on the `net/http` for the distributed tracing.

Reply:
exactly; especially what's happening if the underlying adapter (i.e. `httpclient`) doesn't provide distributed tracing? All three scenarios are plausible and it's reasonable that you want distributed tracing only with `faraday`.

Reply:
What do think about creating then another middleware for the distributed tracing? I think their middleware API works really well for this level of granularity (a lot of people have, for example, a middleware that injects an authentication header in each request). That way the user could do something like:

```rb
::Faraday.new('http://example.com') do |builder|
  builder.use(:dd_instrumentation)
  builder.use(:dd_distributed_tracing)
end
```

Reply:
that sounds definitely legit to me; it would be great to start writing also some documentation for this module so that, while writing the docs, we're aware of how much could be good (or confusing) our ways to handle these 3 cases ^^